src/blob_file_builder.cc

```diff
@@ -5,26 +5,38 @@ namespace titandb {
 
 BlobFileBuilder::BlobFileBuilder(const TitanDBOptions& db_options,
                                  const TitanCFOptions& cf_options,
-                                 WritableFileWriter* file)
+                                 WritableFileWriter* file,
+                                 uint32_t blob_file_version)
     : builder_state_(cf_options.blob_file_compression_options.max_dict_bytes > 0
                          ? BuilderState::kBuffered
                          : BuilderState::kUnbuffered),
       cf_options_(cf_options),
       file_(file),
+      blob_file_version_(blob_file_version),
       encoder_(cf_options.blob_file_compression,
                cf_options.blob_file_compression_options) {
-#if ZSTD_VERSION_NUMBER < 10103
+  status_ = BlobFileHeader::ValidateVersion(blob_file_version_);
+  if (!status_.ok()) {
+    return;
+  }
   if (cf_options_.blob_file_compression_options.max_dict_bytes > 0) {
+    if (blob_file_version_ != BlobFileHeader::kVersion2) {
+      status_ = Status::NotSupported(
+          "dictionary comparession is not supported by blob file version 1");
```

Comment:
```suggestion
          "dictionary compression is not supported by blob file version 1");
```

Reply:
It is a cherry-pick. I will update master instead.